crates/runtime/src/libcalls.rs

```diff
@@ -201,6 +202,33 @@ pub unsafe extern "C" fn wasmtime_imported_memory32_size(
     instance.imported_memory_size(memory_index)
 }
 
+/// Implementation of `table.size`.
+pub unsafe extern "C" fn wasmtime_table_size(vmctx: *mut VMContext, table_index: u32) -> u32 {
+    let table_index = TableIndex::from_u32(table_index);
+    let instance = (&mut *vmctx).instance();
+    let table = instance.get_table(table_index);
+    table.size()
+}
+
+/// Implementation of `table.grow` for `externref`s.
+pub unsafe extern "C" fn wasmtime_table_grow_extern_ref(
+    vmctx: *mut VMContext,
+    table_index: u32,
+    delta: u32,
+    init_value: *mut u8,
+) -> u32 {
+    let instance = (&mut *vmctx).instance();
+    let table_index = TableIndex::from_u32(table_index);
+    let init_value = if init_value.is_null() {
+        None
+    } else {
+        Some(VMExternRef::clone_from_raw(init_value))
```

Comment:
I'm a bit curious about the ownership here. Presumably it's not safe to `gc` at the beginning of this function, right? The `init_value` pointer is in the activations table for sure, but there's no guarantee it's reachable from any stack map?

Reply:
Correct. I should probably add a warning to the docs at the top of this file, like how we warn about traps and potential leaks from skipped `drop`s.